clients/go/admin/token_source_provider.go

```diff
@@ -46,15 +51,24 @@ func NewTokenSourceProvider(ctx context.Context, cfg *Config, tokenCache cache.T
 		}
 
 		scopes := cfg.Scopes
-		if len(scopes) == 0 {
-			clientMetadata, err := authClient.GetPublicClientConfig(ctx, &service.PublicClientAuthConfigRequest{})
+		audienceValue := cfg.Audience
+
+		if len(scopes) == 0 || len(audienceValue) == 0 {
```

Comment:
If I'm understanding correctly with this change now we'll always make the call to get the public client config even in the case where the audience property is not actually required? (It isn't always required, right - this is just custom for some deployments?)

Is there anyway to indicate we want to explicitly fetch the audience property so that end users don't have to now update their deployments with a dummy config value for the audience property to avoid this lookup even when scopes are set?

Reply:
Done Added added new config flag `useAudienceFromAdmin` which will force pull config from Admin only if its set otherwise it wont call the publicEndpoint config